src/app/teams/teams.component.ts

```diff
@@ -44,14 +47,20 @@ export class TeamsComponent implements OnInit {
       catchError(error => {
         this.snackBar.open('Could not load teams: ' + error.error, 'Dismiss');
         console.log(error);
-        return of([])
+        return of(new TeamList([], false));
       })
-    ).subscribe((teams?: Team[]) => {
-      if (teams) {
-        this.teams = teams.map(t => new ImmutableTeam(t));
+    ).subscribe((teamList?: TeamList) => {
+      if (teamList?.teams) {
+        this.teams = teamList.teams.map(t => new ImmutableTeam(t));
       } else {
         this.teams = undefined;
       }
+      if (teamList?.addTeamPermissions) {
```

Comment:
If I understood this correctly, the "AddTeamPermissions" is meant to act as an application level setting for whether the users are allowed to add team(s) or not. Is that true? If so, can you add appropriate documentation in the readme? If not, can you please explain what is the purpose of it?

Reply:
I have tried to explain it in the README, could you let me how I could improve what I've written to make this clearer?

Reply:
As noted above, I think renaming `addTeamPermissions` would really help here.

Reply:
👍🏻